entity-service-change-event-generator/src/main/java/org/hypertrace/entity/service/change/event/impl/EntityChangeEventGeneratorImpl.java

```diff
@@ -147,6 +172,67 @@ private void sendUpdateNotification(
     }
   }
 
+  private boolean shouldSendNotification(Entity prevEntity, Entity currEntity) {
+    boolean isAttributesChanged = false, isCoreEntityChanged = false;
+    for (FieldDescriptor fieldDescriptor : Entity.getDescriptor().getFields()) {
+      switch (fieldDescriptor.getNumber()) {
+        case Entity.ATTRIBUTES_FIELD_NUMBER:
+          isAttributesChanged = checkIfAttributesChanged(prevEntity, currEntity);
+          break;
+        default:
+          boolean isFieldChanged =
+              !prevEntity.getField(fieldDescriptor).equals(currEntity.getField(fieldDescriptor));
+          if (!isCoreEntityChanged) {
+            isCoreEntityChanged = isFieldChanged;
+          }
+          break;
+      }
+    }
+
+    return isCoreEntityChanged || isAttributesChanged;
+  }
+
+  private boolean checkIfAttributesChanged(Entity prevEntity, Entity currEntity) {
+    MapDifference<String, AttributeValue> attributesDiff =
```

Comment:
This can dramatically simplify.
1. Remove the skipped attributes out of both maps before doing the diff, rather than doing it after each direction of diff.
2. We don't actually care what the diff is, just that there is one, so rather than taking added, deleted and changed, you can just say - `prevMap.size == currMap.size == entriesInCommon.size`